drivers/builtin/src/nist_kw.c

```diff
@@ -410,316 +422,43 @@ int mbedtls_nist_kw_unwrap(mbedtls_nist_kw_context *ctx,
          * If Plen is larger than the plaintext and padding, padlen will be
          * larger than 8, because of the type wrap around.
          */
-        padlen = in_len - KW_SEMIBLOCK_LENGTH - Plen;
+        padlen = input_length - KW_SEMIBLOCK_LENGTH - Plen;
         ret = mbedtls_ct_error_if(mbedtls_ct_uint_gt(padlen, 7),
-                                  MBEDTLS_ERR_CIPHER_AUTH_FAILED, ret);
+                                  PSA_ERROR_INVALID_SIGNATURE, ret);
         padlen &= 7;
 
         /* Check padding in "constant-time" */
         const uint8_t zero[KW_SEMIBLOCK_LENGTH] = { 0 };
         diff = mbedtls_ct_memcmp_partial(
-            &output[*out_len - KW_SEMIBLOCK_LENGTH], zero,
+            &output[*output_length - KW_SEMIBLOCK_LENGTH], zero,
             KW_SEMIBLOCK_LENGTH, KW_SEMIBLOCK_LENGTH - padlen, 0);
 
         if (diff != 0) {
-            ret = MBEDTLS_ERR_CIPHER_AUTH_FAILED;
+            ret = PSA_ERROR_INVALID_SIGNATURE;
         }
 
-        if (ret != 0) {
+        if (ret != PSA_SUCCESS) {
             goto cleanup;
         }
         memset(output + Plen, 0, padlen);
-        *out_len = Plen;
+        *output_length = Plen;
     } else {
-        ret = MBEDTLS_ERR_CIPHER_FEATURE_UNAVAILABLE;
+        ret = PSA_ERROR_NOT_SUPPORTED;
         goto cleanup;
     }
 
 cleanup:
-    if (ret != 0) {
-        memset(output, 0, *out_len);
-        *out_len = 0;
+    if (ret != PSA_SUCCESS  && output != NULL) {
+        memset(output, 0, *output_length);
```

Comment:
There is a buffer overflow here if the value of `*output_length` on entry happens to be large. The obvious fix would be to set `*output_length = 0` earlier, before the first `goto cleanup`.